timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/LeadershipComponents.java

```diff
@@ -54,15 +65,44 @@ public class LeadershipComponents {
         this.healthCheckPingers = healthCheckPingers;
     }
 
-    public <T> T wrapInLeadershipProxy(Client client, Class<T> clazz, Supplier<T> delegateSupplier) {
+    private <T> T wrapInLeadershipProxy(
+            Class<T> clazz, Supplier<T> delegateSupplier, LeadershipCoordinator leadershipCoordinator) {
+        T instance = AwaitingLeadershipProxy.newProxyInstance(clazz, delegateSupplier, leadershipCoordinator);
+        // this is acceptable since the proxy returned implements Closeable and needs to be closed
+        closer.register((Closeable) instance);
+        return instance;
+    }
+
+    public LeadershipServices createServices(
```

Comment:
I'd call the result class something different; they are technically not leadership services but services wrapped with leadership proxies / checks

Something like "LeadershipAwareTimeLockServices" or something shorter. The method could still be called `wrapInLeadershipProxy` with the things being wrapped being the suppliers given

Reply:
How about `LeadershipProxies`? I was going for something short yet descriptive as it isn't really exposed outside the module itself but crosses package boundary.

Reply:
updated